pdbtools/pdb_selres.py

```diff
@@ -164,24 +165,35 @@ def _validate_opt_range(value, resid_list):
             sys.exit(1)
 
         option = args[0][1:]
-        fh = open(args[1], 'r')
+        fh = open(args[1], 'rb')
 
     else:  # Whatever ...
         sys.stderr.write(__doc__)
         sys.exit(1)
 
     # Read file handle to extract residue numbers
+    # Because sys.stdin is not seekable we store the
+    # lines again in an iterator.
+    buffer = iter([])
     resid_list = []
+
     records = ('ATOM', 'HETATM', 'TER', 'ANISOU')
     prev_res = None
     for line in fh:
+        line = line.decode('utf-8')
```

Comment:
I believe it would be faster if you remove this line (183) from here.

Set records to bytes:

```python
records = (b'ATOM', b'HETATM', b'TER', b'ANISOU')
```

And decode only when adding to the list:

```python
buffer = iter_chain(buffer, [line.decode('utf-8')])
```

Reply:
Good catch!